### PR TITLE
executor,planner: Fix tiflash plan with limit operator may lose execution summary problem

### LINKED ISSUE #50455: tiflash explain analyze actRows is zero unexpectly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. load tpch50 dataset(may be other smaller dataset is also ok)
2. run sql:
```
set @@tidb_isolation_read_engines = 'tiflash';
explain analyze select * from lineitem inner join orders on lineitem.l_orderkey = orders.o_orderkey limit 1000;
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
actRow is not zero
### 3. What did you see instead (Required)
actRows of tiflash executor is zero.

<img width="1917" alt="image" src="https://github.com/pingcap/tidb/assets/7493273/7804c4ad-3123-40f4-8d20-baebbd4f81c4">

### 4. What is your TiDB version? (Required)
```
MySQL [test]> select tidb_version();
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                     |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v7.6.0-alpha-602-gfbe232e3f8
Edition: Community
Git Commit Hash: fbe232e3f85d1c173ad15fe2401c09af4b1424de
Git Branch: HEAD
UTC Build Time: 2024-01-16 05:14:57
GoVersion: go1.21.0
Race Enabled: false
Check Table Before Drop: false
Store: tikv |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.044 sec)
```
<!-- Paste the output of SELECT tidb_version() -->

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #50455 

Problem Summary:
Currently, if limit operator is not pushed down to tiflash side, then execution summary in tiflash side will be lost. This is because we only check if limit operator exists under TableReader. Refine the logic so that it checks if limit operator exists in the path from TableReader to Root.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Manually verified explain analyze insert/update/delete statements' execution summary.
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/internal/mpp/local_mpp_coordinator.go [hunk 1]
```diff
@@ -336,14 +339,18 @@ func (c *localMppCoordinator) fixTaskForCTEStorageAndReader(exec *tipb.Executor,
...
 	switch x := plan.(type) {
 	case *plannercore.PhysicalLimit:
-		return true
+		return needReportExecutionSummary(x.Children()[0], destTablePlanID, true)
+	case *plannercore.PhysicalTableReader:
+		if foundLimit {
+			return x.GetTablePlan().ID() == destTablePlanID
+		}
 	default:
 		for _, child := range x.Children() {
```
Comment (guo-shaoge) [inline pkg/executor/internal/mpp/local_mpp_coordinator.go:352]: I'm not sure if CTE also works. Its children will not be returned by calling `Children()`

Comment (yibin87) [reply pkg/executor/internal/mpp/local_mpp_coordinator.go:352]: OK, I'll check it.

**Review by windtalker — APPROVED** [review]
lgtm

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**